### PR TITLE
libvirt_mem: Fix the file nonexistent and adapt x86_64

### PR DESCRIPTION
**Pull request by hsmj1412**
Signed-off-by: Junxiang Li <junli@redhat.com>

### CODE REVIEW

**Comment by hsmj1412** [comment]
The reason for the test failure is the machine without numa nodes, so it is a expected failure
avocado run --vt-type libvirt --vt-machine-type i440fx libvirt_mem.positive_test --vt-connect-uri qemu:///system
JOB ID     : 80298ac39c698f1677dcff18f7fe1b58686fafc5
JOB LOG    : /root/avocado/job-results/job-2018-02-26T20.24-80298ac/job.log
 (01/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.mem_basic: PASS (50.66 s)
 (02/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.memory.hot.plug: PASS (67.04 s)
 (03/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.memory.hot.unplug: PASS (70.37 s)
 (04/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.memory.cold.plug: PASS (49.68 s)
 (05/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.memory.cold.unplug: PASS (51.55 s)
 (06/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.no_attach: PASS (50.74 s)
 (07/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.with_source: PASS (50.48 s)
 (08/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.no_numatune.hot_plug: PASS (89.34 s)
 (09/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.no_numatune.hot_unplug: PASS (92.31 s)
 (10/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.no_numatune.cold_plug: PASS (71.67 s)
 (11/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.with_numatune.hot_plug: PASS (88.34 s)
 (12/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.with_numatune.hot_unplug: PASS (92.13 s)
 (13/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.hugepages.with_numatune.cold_plug: PASS (70.92 s)
 (14/14) type_specific.io-github-autotest-libvirt.libvirt_mem.positive_test.without_numa_save_restore: FAIL (6.47 s)
RESULTS    : PASS 13 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 905.79 s

File: libvirt/tests/src/libvirt_mem.py [hunk 1]
```diff
@@ -479,18 +471,34 @@ def modify_domain_xml():
...
-                libvirt.check_exit_status(ret, detach_error)
+                try:
+                    libvirt.check_exit_status(ret, detach_error)
+                except Exception as detail:
+                    dmesg_file = tempfile.mktemp(dir=data_dir.get_tmp_dir())
+                    try:
+                        session = vm.wait_for_login()
+                        utils_misc.verify_dmesg(dmesg_log_file=dmesg_file, ignore_result=True, session=session,
+                                                level_check=5)
+                        session.close()
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_mem.py]: move this line to ahead of line 484 ensure that session.close() can be called whenever exception happen

File: libvirt/tests/src/libvirt_mem.py [hunk 2]
```diff
@@ -479,18 +471,34 @@ def modify_domain_xml():
...
+                    except Exception:
+                        test.fail("After memory unplug Unable to connect to VM"
+                                  " or unable to collect dmesg")
+                    if os.path.exists(dmesg_file):
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_mem.py]: add session.close() before this line

File: libvirt/tests/src/libvirt_mem.py [hunk 3]
```diff
@@ -479,18 +471,34 @@ def modify_domain_xml():
...
+                        if not re.findall(r'memory memory\d+?: Offline failed', open(dmesg_file, 'r').read()):
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_mem.py]: add one comment why we need grep 'memory information' in dmesg file

File: libvirt/tests/src/libvirt_mem.py [hunk 4]
```diff
@@ -479,18 +471,34 @@ def modify_domain_xml():
...
+                            os.remove(dmesg_file)
+                            test.fail(detail)
+                        unplug_failed_with_known_error = True
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_mem.py]: I don't think we need this statement: unplug_failed_with_known_error = True

File: libvirt/tests/src/libvirt_mem.py (hunk 1 above)
Comment (hsmj1412) [reply libvirt/tests/src/libvirt_mem.py]: done

File: libvirt/tests/src/libvirt_mem.py (hunk 2 above)
Comment (hsmj1412) [reply libvirt/tests/src/libvirt_mem.py]: done

File: libvirt/tests/src/libvirt_mem.py (hunk 3 above)
Comment (hsmj1412) [reply libvirt/tests/src/libvirt_mem.py]: done

**Pull request merged**